ApplicationCommands/src/Commands/ManagePermissions.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DisCatSharp.ApplicationCommands;
+using DisCatSharp.Entities;
+using DisCatSharp.Exceptions;
+
+namespace DisCatSharp.Examples.ApplicationCommands.Commands
+{
+    public class ManagePermissions : ApplicationCommandsModule
+    {
+        [SlashCommandGroup("manage_permissions", "Allows to add/remove permissions to execute commands to users/roles", false)]
+        public class Command : ApplicationCommandsModule
+        {
+            [SlashCommand("add_user", "Add permission to user.")]
+            public static async Task UserAdd(InteractionContext context, [Option("user", "User to be added")] DiscordUser user,
+                [Option("command", "The command to which we must give access")] CommandNames commandName)
+            {
+                DiscordInteractionResponseBuilder discordInteractionResponseBuilder = new() {IsEphemeral = true};
+
+                DiscordApplicationCommand command = GetCommand(context, commandName.ToString());
+
+                List<DiscordApplicationCommandPermission> perms = new();
+
+                try
+                {
+                    // Get all current command permissions
+                    perms = (await context.Client.GetApplicationCommandPermissionAsync(context.Guild.Id, command.Id))
+                        .Permissions.ToList();
+                }
+                catch (NotFoundException)
+                {
+                    // Ignore if permissions not found
+                }
+
+                // Add user to the permissions
+                perms.Add(new(user.Id, ApplicationCommandPermissionType.User, true));
+
+                // Overwrite permissions
+                await context.Client.OverwriteGuildApplicationCommandPermissionsAsync(context.Guild.Id, command.Id, perms);
+                
+                await context.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, discordInteractionResponseBuilder.WithContent("User added successfully!"));
+            }
+            
+            [SlashCommand("add_role", "Add permission to role.")]
+            public static async Task RoleAdd(InteractionContext context, [Option("role", "Role to be added")] DiscordRole role,
+                [Option("command", "The command to which we must give access")] CommandNames commandName)
+            {
+                DiscordInteractionResponseBuilder discordInteractionResponseBuilder = new() {IsEphemeral = true};
+
+                var command = GetCommand(context, commandName.ToString());
+
+                List<DiscordApplicationCommandPermission> perms = new();
+
+                try
```

Comment:
Checks should be done here. Making sure we own the command and if the command doesn't exist, telling them so.